Subsystems/cmdGui/CommandSystem.py

```diff
@@ -17,227 +17,104 @@
-           cmd_args = shlex.split(launch_string)
-           subprocess.Popen(cmd_args)
-
-    #
-    # Quick button wrappers
-    #
-    def QuickButton_1(self):
-        self.ProcessQuickButton(0)
-    def QuickButton_2(self):
-        self.ProcessQuickButton(1)
-    def QuickButton_3(self):
-        self.ProcessQuickButton(2)
-    def QuickButton_4(self):
-        self.ProcessQuickButton(3)
-    def QuickButton_5(self):
-        self.ProcessQuickButton(4)
-    def QuickButton_6(self):
-        self.ProcessQuickButton(5)
-    def QuickButton_7(self):
-        self.ProcessQuickButton(6)
-    def QuickButton_8(self):
-        self.ProcessQuickButton(7)
-    def QuickButton_9(self):
-        self.ProcessQuickButton(8)
-    def QuickButton_10(self):
-        self.ProcessQuickButton(9)
-    def QuickButton_11(self):
-        self.ProcessQuickButton(10)
-    def QuickButton_12(self):
-        self.ProcessQuickButton(11)
-    def QuickButton_13(self):
-        self.ProcessQuickButton(12)
-    def QuickButton_14(self):
-        self.ProcessQuickButton(13)
-    def QuickButton_15(self):
-        self.ProcessQuickButton(14)
-    def QuickButton_16(self):
-        self.ProcessQuickButton(15)
-    def QuickButton_17(self):
-        self.ProcessQuickButton(16)
-    def QuickButton_18(self):
-        self.ProcessQuickButton(17)
-    def QuickButton_19(self):
-        self.ProcessQuickButton(18)
-    def QuickButton_20(self):
-        self.ProcessQuickButton(19)
-    def QuickButton_21(self):
-        self.ProcessQuickButton(20)
-    def QuickButton_22(self):
-        self.ProcessQuickButton(21)
-
+        if cmdPageIsValid[idx]:
+            lePID = getattr(self, f'lineEditPktId_{idx}')
+            leAddr = getattr(self, f'lineEdit_{idx}')
+            pktId = lePID.text()
+            address = leAddr.text()
+            launch_string = (
+                f'python3 {ROOTDIR}/{cmdClass[0]} '
+                f'--title=\"{cmdPageDesc[idx]}\" --pktid={pktId} '
+                f'--file={cmdPageDefFile[idx]} --address=\"{address}\" '
+                f'--port={cmdPagePort[idx]} --endian={cmdPageEndian[idx]}')
+            cmd_args = shlex.split(launch_string)
+            subprocess.Popen(cmd_args)
 
     #
     # Determines if command requires parameters
     #
-    def checkParams(self, idx):
-        pickle_file = 'ParameterFiles/' + quickParam[idx]
+    @staticmethod
+    def checkParams(idx):
+        pickle_file = f'{ROOTDIR}/ParameterFiles/{quickParam[idx]}'
         try:
-            with open(pickle_file,'rb') as pickle_obj:
-                dataTypesOrig, paramNames, paramLen, paramDesc, dataTypesNew, stringLen = pickle.load(pickle_obj)
-                if len(paramNames) > 0: # if has parameters
-                   return True
-                else:
-                   return False
+            with open(pickle_file, 'rb') as pickle_obj:
+                _, paramNames, _, _, _, _ = pickle.load(pickle_obj)
```

Comment:
minor quibble, I'd rather this be "paramNames = pickle.load(pickle_obj)[1]"

Reply:
Ah, yes. I see what you mean. I can do that